odoo/addons/base/models/ir_model.py

```diff
@@ -2627,8 +2609,7 @@ def _process_end(self, modules):
     @api.model
     def toggle_noupdate(self, model, res_id):
         """ Toggle the noupdate flag on the external id of the record """
-        record = self.env[model].browse(res_id)
-        if record.check_access_rights('write'):
+        if self.env[model].check_access('write'):
```

Comment:
Shouldn't you `browse`? I think that you should have access on the record to change its "noupdate" flag and this was probably incorrectly checked before.